qutip/core/data/base.pxd

```diff
@@ -1,10 +1,9 @@
 #cython: language_level=3
 
-import numpy as np
-cimport numpy as cnp
+cdef extern from "src/intdtype.h":
+    # cython is smart enough to understand this int can be 32 or 64 bits.
+    ctypedef int idxint
```

Comment:
In `intdtype.h` we can do:
```c
#define QUTIP_INT_SIZE 32
```
and then inside this `cdef extern` block add:
```c
int QUTIP_INT_SIZE
```
Of course `QUTIP_INT_SIZE` is not really an integer, but rather a constant, but Cython only cares about the type (i.e. that the constant can be used where an int can be used).

This will take away the need for the cunning `size_check` logic and will also make setting the value in `qutip.settings` simpler if we expose it.